src/components/Articles/VoteButton.jsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState } from "react"
+import axiosInstance from "../utilities/axios"
+
+const VoteButton = ({ articleId, votes, setVotes }) => {
+  const [loading, setLoading] = useState(false)
+
+  const handleVote = async (increment) => {
+    try {
+      setLoading(true)
+
+      // optimistic
+      setVotes((prevVotes) => prevVotes + (increment ? 1 : -1))
+
+      //   await axiosInstance.patch(`/articles/${articleId}`, { increment })
+
+      setLoading(false)
+    } catch (error) {
+      console.error("Error voting:", error)
+
+      setVotes((prevVotes) => prevVotes - (increment ? 1 : -1))
+
+      alert("Failed to vote. Please try again later.")
```

Comment:
best to avoid using alerts to let users know about errors, some browsers don't allow them
Consider using something like Toast.js instead
In this case visual feedback like the vote being reset would be enough of an indication something being wrong